### PR TITLE
Add decomposition for transpose_copy

### PR DESCRIPTION
**Pull request by rec**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130943

* Extracted from #128416

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -19746,6 +19746,26 @@ def sample_inputs_alias_copy(op_info, device, dtype, requires_grad, **kwargs):
            # vmap does not support inplace views
            check_inplace_batched_forward_grad=False,
            sample_inputs_func=sample_inputs_transpose_swapdims),
+    OpInfo('transpose_copy',
+           aliases=('swapdims', 'swapaxes'),
```
Comment (peterbell10) [inline torch/testing/_internal/common_methods_invocations.py:19750]: I don't think these apply, and thats probably why the jit alias mapping tests are failing.

**Comment by rec** [comment]
topic: not user facing

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (rec) [reply torch/testing/_internal/common_methods_invocations.py:19750]: Removed! And also removed one of the skips.  The other one continues to occur, and is in the same class as in the `squeeze_copy` decomposition pull request, https://github.com/pytorch/pytorch/pull/130941

**Review by amjames — APPROVED** [review]

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**